.changeset/quiet-shirts-trade.md

```diff
@@ -0,0 +1,9 @@
+---
+'@lit-labs/nextjs': minor
+---
+
+Add support for Next.js 14 and App Router. No longer supports Next.js 12.
+
+Note: Components in the App Router are by default React Server Components. Deep SSR of Lit components does **not** work within server components as they result in React hydration mismatch due to the presence of the `<template>` element in React Flight serialized server component data, and the custom element definitions will not be included with the client bundle either.
```

Comment:
"Flight" appears to be React/Next team's code word for the system/format behind server component serialization, serving/parsing. I found reference to this name in comments and functions as I was diving through their codebases, e.g. https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/use-flight-response.tsx

This blog post explains it. https://timtech.blog/posts/react-server-components-rsc-no-framework/#react-server-%26-react-flight-apis

But there's no official doc that mentions it by name. Only react/next team members mentioning it in comments and PRs like https://github.com/vercel/next-react-server-components/issues/13#issuecomment-988438962

Since many would wonder what that means the same as you, I'll remove "Flight" and just call them RSC payloads or something like that.